airflow/providers/amazon/aws/operators/ecs.py

```diff
@@ -174,9 +215,21 @@ def execute(self, context: Context):
         cluster_state = cluster_details.get("status")
 
         if cluster_state == EcsClusterStates.INACTIVE:
-            # In some circumstances the ECS Cluster is deleted immediately,
-            # so there is no reason to wait for completion.
+            # if the cluster doesn't have capacity providers that are associated with it,
+            # the deletion is instantaneous, and we don't need to wait for it.
```

Comment:
`cluster_details` has the `capacityProviders` associated with the nodegroup. Would that be a better way to decide whether we want to wait for completion or not?

Reply:
hmm, we could do that, but the check on the status above is already taking care of that. We can write a different check, but the result would be the same.